src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/judgeapproval/ApproveDraftOrdersAboutToStartHandlerTest.java

```diff
@@ -0,0 +1,665 @@
+
+    @Mock
+    private HearingService hearingService;
+
+    @InjectMocks
+    private ApproveDraftOrdersAboutToStartHandler handler;
+
+    @Test
+    void canHandle() {
+        assertCanHandle(handler, CallbackType.ABOUT_TO_START, CaseType.CONTESTED, EventType.APPROVE_ORDERS);
+    }
+
+    @Test
+    void givenUserHasHearingsReadyToReview_whenHandle_thenReturnSortedHearings() {
+        FinremCaseData caseData = new FinremCaseData();
+
+        DraftOrderDocumentReview document1 = DraftOrderDocumentReview.builder().orderStatus(TO_BE_REVIEWED)
+            .build();
+        DraftOrderDocumentReview document2 = DraftOrderDocumentReview.builder().orderStatus(APPROVED_BY_JUDGE)
+            .build();
+
+        DraftOrderDocReviewCollection collectionItem1 = new DraftOrderDocReviewCollection(document1);
+        DraftOrderDocReviewCollection collectionItem2 = new DraftOrderDocReviewCollection(document2);
+
+        DraftOrdersReview review1 = DraftOrdersReview.builder()
+            .hearingDate(LocalDate.of(2024, 8, 6))
+            .hearingType("Hearing Type 1")
+            .hearingJudge("Judge 1")
+            .hearingTime("09:00 A.M.")
+            .draftOrderDocReviewCollection(List.of(collectionItem1, collectionItem2))
+            .build();
+
+        DraftOrdersReviewCollection reviewCollection1 = new DraftOrdersReviewCollection(review1);
+        caseData.getDraftOrdersWrapper().setDraftOrdersReviewCollection(List.of(reviewCollection1));
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(
+            FinremCallbackRequestFactory.from(1727874196328932L, caseData), AUTH_TOKEN);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getErrors()).isEmpty();
+    }
+
+    @Test
+    void givenUserHasPsaReadyToReview_whenHandle_thenReturnSortedHearings() {
+        FinremCaseData caseData = new FinremCaseData();
+        PsaDocumentReview document1 = PsaDocumentReview.builder().orderStatus(TO_BE_REVIEWED)
+            .build();
+        PsaDocReviewCollection psaCollectionItem1 = new PsaDocReviewCollection(document1);
+
+        DraftOrdersReview review1 = DraftOrdersReview.builder()
+            .hearingDate(LocalDate.of(2024, 8, 6))
+            .hearingType("Hearing Type 1")
+            .hearingJudge("Judge 1")
+            .hearingTime("09:00 A.M.")
+            .psaDocReviewCollection(List.of(psaCollectionItem1))
+            .build();
+
+        DraftOrdersReviewCollection reviewCollection1 = new DraftOrdersReviewCollection(review1);
+        caseData.getDraftOrdersWrapper().setDraftOrdersReviewCollection(List.of(reviewCollection1));
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(
+            FinremCallbackRequestFactory.from(1727874196328932L, caseData), AUTH_TOKEN);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getErrors()).isEmpty();
+    }
```

Comment:
Both tests above look fine.  Method name suggests you may have intended to check the sorting.

Mentioning in case you did.

Reply:
I didn't see an explicit sorting test.  Sorry if I missed it.

If you wanted to test sorting, I liked your CSV approach in HearingServiceTest.  You could check the date ordering was right, without actually using your compareTo.

Reply:
ah.. those are old test cases. I removed the first page for selecting hearing. And those test cases should be removed .